src/main/java/decodes/tsdb/WaterTempProfiles.java

```diff
@@ -0,0 +1,101 @@
+package decodes.tsdb;
+
+import decodes.db.Constants;
+import decodes.db.Site;
+import decodes.db.SiteName;
+import opendcs.dai.TimeSeriesDAI;
+import org.opendcs.utils.FailableResult;
+
+import java.util.ArrayList;
+import java.util.Date;
+
+public class WaterTempProfiles extends DataCollection {
+
+    private TimeSeriesDAI timeSeriesDAO = null;
+
+    /** The time series */
+    private ArrayList<CTimeSeries> tseries;
+
+    /** Handle storing tasklist Record Ranges for computation processor. */
+    private RecordRangeHandle rrHandle;
+
+    private double startDepth;
+    private double increment;
+
+    /** Constructor -- builds an empty collection with a null handle. */
+    public WaterTempProfiles(TimeSeriesDAI DAO, double start, double incr){
+        tseries = new ArrayList<CTimeSeries>();
+        rrHandle = null;
+        timeSeriesDAO = DAO;
+        startDepth = start;
+        increment = incr;
+    }
+    public WaterTempProfiles(ArrayList<CTimeSeries> profiles, TimeSeriesDAI DAO, double start, double incr){
+        tseries = profiles;
+        rrHandle = null;
+        timeSeriesDAO = DAO;
+        startDepth = start;
+        increment = incr;
+    }
+
+    public WaterTempProfiles(TimeSeriesDAI DAO, String wtpId, Date since, Date until, double start, double incr) throws DbIoException, NoSuchObjectException {
+        tseries = new ArrayList<CTimeSeries>();
+        rrHandle = null;
+        timeSeriesDAO = DAO;
+        startDepth = start;
+        increment = incr;
+        boolean loading = true;
+        double currentDepth = startDepth;
+        TimeSeriesIdentifier tsid;
+        tsid = timeSeriesDAO.getTimeSeriesIdentifier(wtpId);
+        while(loading){
+            try {
+                TimeSeriesIdentifier newtsid = tsid.copyNoKey();
```

Comment:
For consideration:

The cwms schema "profiles" use a separate time series to control depth of measurement, or height for atmospheric instead of embedding the value in the name. It's not really used yet so it makes sense that NWDM wasn't doing thing that way but it likely makes sense for this effort to follow that scheme. 

However as I type that out I'm not entirely sure how to describe things lining up other than having multiple measurements within the hour, which makes complete sense for actually measured profiles with the sensor moving but these are calculated all at once.


Keep this as is for now, I'll investigate the profiles more to see how it relates.

Reply:
We established that these profiles are not the same as our "time series" profiles.